sympy/ntheory/generate.py

```diff
@@ -142,22 +142,30 @@ def extend_to_no(self, i):
         while len(self._list) < i:
             self.extend(int(self._list[-1] * 1.5))
 
-    def primerange(self, a, b):
-        """Generate all prime numbers in the range [a, b).
+    def primerange(self, a, b=None):
+        """Generate all prime numbers in the range [a, b) when both a and b are
+           provided and generates all prime numbers till a when only a is provided.
```

Comment:
When both `a` and `b` are there  `a` is taken as included lower limit and `b` as excluded upper limit in range.
But these changes make `a` upper limit when only `a` is provided, which seems wrong.
I think `a` should be made optional so that behaviour of variables remains same.

Reply:
If we make `a` optional then, following the python syntax that an argument right to a default argument must be a default argument we will be forced to take input first `b` then `a` and the user will have to then provide the upper limit first to hold the behavior of the variable 'b'.
Ig making `b` as an optional argument is not changing the behavior of the variables much as in the first line it checks if `b` is provided or not, and if `b` is not provided then `b` is assigned the value if the provided number that is the upper limit and `a` is assigned as 2 that act as the lower limit. So by the end of first if statement `a` is the lower limit and `b` is the upper limit thus not changing the behavior of the variables.

Reply:
This is very confusing as written, as other commenters have noted. How about something like

```
Generate all prime numbers in the range [a, b).

If only one argument is given, `primerange(n)` generates all prime numbers in the range [2, n).
```

The technical detail about the argument `a` becoming the second argument are confusing and shouldn't be noted. As far as the user is concerned, the function has two signatures, `primerange(a, b)` and `primerange(b)`. See also how the Python function `range` is documented (from help(range)):

```
class range(object)
 |  range(stop) -> range object
 |  range(start, stop[, step]) -> range object
```

Also "till a" is ambiguous as to whether or not a is included, and "till" is awkward to use in mathematical contexts. It's better to be clear by using interval notation.